client/components/mma/cancel/cancellationSaves/supporterplus/SupporterPlusOfferReview.tsx

```diff
@@ -0,0 +1,267 @@
+
+const yourOfferBoxCss = css`
+	background-color: #fbf6ef;
+	padding: ${space[4]}px ${space[6]}px;
+	display: flex;
+	flex-direction: column;
+	h4 {
+		${textSansBold17};
+		margin: 0;
+	}
+	${from.desktop} {
+		flex-direction: row;
+		gap: 1ch;
+	}
+`;
+
+const strikethroughPriceCss = css`
+	${textSans17};
+	color: ${neutral[46]};
+	margin: 0;
+`;
+
+const whatsNextTitleCss = css`
+	${textEgyptianBold17};
+	margin-top: ${space[6]}px;
+	${from.desktop} {
+		margin-top: ${space[8]}px;
+	}
+`;
+
+const whatsNextListCss = css`
+	${textEgyptian17};
+	padding: 0;
+	padding-inline-start: 14px;
+	li + li {
+		margin-top: ${space[3]}px;
+	}
+`;
+
+const buttonsCtaHolder = css`
+	margin: ${space[8]}px 0 ${space[6]}px;
+	display: flex;
+	flex-direction: column;
+	gap: ${space[2]}px;
+	${from.phablet} {
+		flex-direction: row;
+		gap: ${space[6]}px;
+		margin-top: ${space[9]}px;
+	}
+`;
+
+const ctaBtnCss = css`
+	width: 100%;
+	justify-content: center;
+	${from.desktop} {
+		width: fit-content;
+	}
+`;
+
+const termsCss = css`
+	${textSans12};
+	color: ${palette.neutral[46]};
+	margin-top: ${space[3]}px;
+`;
+
+export const SupporterPlusOfferReview = () => {
+	const location = useLocation();
+	const routerState = location.state as RouterSate;
+	const navigate = useNavigate();
+
+	const cancellationContext = useContext(
+		CancellationContext,
+	) as CancellationContextInterface;
+
+	const productDetail = cancellationContext.productDetail;
+	const mainPlan = getMainPlan(productDetail.subscription);
+
+	const offerPeriodWord = number2words(routerState.upToPeriods);
+	const offerPeriodType = routerState.upToPeriodsType.toLowerCase();
+	const firstDiscountedPaymentDate = parseDate(
+		routerState.firstDiscountedPaymentDate,
+		'yyyy-MM-dd',
+	).dateStr(DATE_FNS_LONG_OUTPUT_FORMAT);
+	const nextNonDiscountedPaymentDate = parseDate(
+		routerState.nextNonDiscountedPaymentDate,
+		'yyyy-MM-dd',
+	).dateStr(DATE_FNS_LONG_OUTPUT_FORMAT);
+
+	const [performingDiscountStatus, setPerformingDiscountStatus] =
+		useState<OfferApiCallStatus>('NOT_READY');
+
+	const confirmBtnIconProps: {
+		icon?: ReactElement;
+		iconSide?: 'right';
+		disabled?: true;
+		['aria-disabled']?: true;
+	} = {};
+	if (performingDiscountStatus === 'PENDING') {
+		confirmBtnIconProps.icon = <SvgSpinner size="xsmall" />;
+		confirmBtnIconProps.iconSide = 'right';
+		confirmBtnIconProps.disabled = true;
+		confirmBtnIconProps['aria-disabled'] = true;
+	}
+
+	const handleConfirmClick = async () => {
+		setPerformingDiscountStatus('PENDING');
+
+		try {
+			const response = await fetchWithDefaultParameters(
+				'/api/discounts/apply-discount',
+				{
+					method: 'POST',
+					body: JSON.stringify({
+						subscriptionNumber:
+							productDetail.subscription.subscriptionId,
+					}),
+				},
+			);
+
+			if (response.ok) {
+				navigate('../offer-confirmed', {
+					state: routerState,
+				});
+			} else {
+				setPerformingDiscountStatus('FAILED');
```

Comment:
Maybe one for the future (and totally just my opinion) - if a non-okay response is always treated the same way as an error I think it can be a nice pattern to put that logic close to the `fetch` call - i.e. return a rejected promise if the response isn't okay. I think it can simplify the calling code, here we wouldn't need the condition in the try block, the `catch` block would handle both types of errors so the `setPerformingDiscountStatus('FAILED');` wouldn't need to be repeated.

Reply:
yeah I think I agree, although maybe this can be addressed in a subsequent pr

Reply:
yeah absolutely, not one for now 👍